src/Sylius/Bundle/ApiBundle/Resources/config/serialization/Commands/AddProductReview.xml

```diff
@@ -28,7 +28,7 @@
             <group>shop:product_review:create</group>
             <group>shop:product_review:read</group>
         </attribute>
-        <attribute name="productCode">
+        <attribute name="product">
```

Comment:
I'm not sure whether it would work, but have you tried `serialized-name`?
```suggestion
        <attribute name="productCode" serialized-name="product">
```